### ISSUE TITLE
Unable to load local dependency

### ISSUE THREAD

**Issue by OxSon** [sigmaSd/IRust#145]
I have installed `irust` on MacOS sonoma 14.6.1 and attempted to follow the instructions in [the irust book's "interactive usage" section](https://sigmasd.github.io/irust_book/interactive-usage/).

Expected behavior: able to add local project by path, import from `lib.rs` and use defined pub functions.

Actual output below (with personal information removed and indicated with `...snip...`:

```
In: :add --path .
 Adding dep [-]
      Adding lis (local) to dependencies
     Locking 1 package to latest compatible version
      Adding lis v0.1.0 (/Users/...snip.../lis)


In: use lis::*;
IRust: failed to format the error output.
This is a bug in IRust.
Feel free to open a bug-report at https://github.com/sigmaSd/IRust/issues/new with the next text:

original_output:
error: failed to get `lis` as a dependency of package `irust_host_repl_15096e4a63114a1d94ef24c8fdbf2d1e v0.1.0 (/private/var/folders/zp/crj3b1gx1zxgjtlbw_dg3wlm0000gn/T/irust_repls/irust_host_repl_15096e4a63114a1d94ef24c8fdbf2d1e)`

Caused by:
  failed to load source for dependency `lis`

Caused by:
  Unable to update /private/Users/...snip.../lis

Caused by:
  failed to read `/private/Users/...snip.../lis/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
```

Ref #131 for previous related issue. Based on what has been said in that issue, I think it's possible the correct path should `/private/tmp/...`?

**Comment by sigmaSd** [comment]
I retsted on linux and it does work
![image](https://github.com/user-attachments/assets/591e2e5e-a893-4bab-833e-5305ef29014f)

This issue and the other one seems to be related to mac, I don't have a mac to test, but I'll try to figure it out
My first hunch is this is something to do with permissions and access to certain paths in macos

**Comment by sigmaSd** [comment]
Can you try this workaround:
- create a directory somewhere (outside of private, just somewhere in your normal home path)
- run irust but make it use that new directory as tmp directory `IRUST_TEMP_DIR=ABSOLUTE_PATH_TO_THAT_FOLDER irust`
- test adding a local library again

**Comment by sigmaSd** [comment]
fixed in 1.71.29

**Comment by OxSon** [comment]
I was on version 1.71.27 (came from a plain `cargo install irust`). I uninstalled and did `cargo install irust --version 1.71.29` and the problem is fixed.
